src/VisualStudio/Core/Impl/ProjectSystem/CPS/TempPECompiler.cs

```diff
@@ -37,22 +40,27 @@ public async Task<bool> CompileAsync(IWorkspaceProjectContext context, string ou
                 throw new ArgumentException(nameof(outputFileName), "Must specify an output file name.");
             }
 
-            var project = _workspace.CurrentSolution.GetProject(context.Id);
+            var project = _workspace.CurrentSolution.GetRequiredProject(context.Id);
+
+            // Start by fetching the compilation we have already have that will have references correct
+            var compilation = await project.GetRequiredCompilationAsync(cancellationToken).ConfigureAwait(false);
 
-            // Remove all files except the ones we care about
-            var documents = project.Documents;
-            foreach (var document in documents)
+            // Update to just the syntax trees we need to keep
+            var syntaxTrees = new List<SyntaxTree>(capacity: filesToInclude.Count);
```

Comment:
ArrayBuilder?

Reply:
Won't change since this isn't a high-traffic path. (Not sure its worth holding up the fix for it.)